src/control/cmd/daos/pool.go

```diff
@@ -329,6 +329,9 @@ func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.Poo
 		if queryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
 			poolInfo.DisabledRanks = rs
 		}
+		if queryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
+			poolInfo.SuspectRanks = rs
```

Comment:
You haven't added the SuspectEngines option to the check up at line 302, so `rlPtr` will always be `nil`.

I know that you're just following the pattern that was here before (which came from a quick refactoring to support pool queries from different parts of the `daos` tool code), but it would be ideal to rework `queryPool()` so that a caller can invoke it once with all of the desired bits set, and then inside of `queryPool()` we can take care of performing multiple pool queries to get the various rank lists. It shouldn't be up to callers of `queryPool()` to do all of this work to build the full response.

Longer-term, I think we need to add a new client-side pool query API that allows all of these rank lists to be fetched at once, similar to how `ds_mgmt_pool_query()` works.

Reply:
Thanks @mjmac I tried to refactor queryPool(), I agreed with longer-term, let's address that in the separate PR.